python/dbsp/config.py

```diff
@@ -17,92 +18,169 @@
+from dbsp_api_client.models.attached_connector import AttachedConnector
+from dbsp_api_client.models.direction import Direction
+from dbsp_api_client.models.connector_type import ConnectorType
 from dbsp_api_client.models.csv_encoder_config import CsvEncoderConfig
 from dbsp_api_client.api.config import new_config
 from dbsp_api_client.api.config import update_config
 from dbsp_api_client.api.pipeline import new_pipeline
+from dbsp_api_client.api.connector import new_connector
 from dbsp.pipeline import DBSPPipeline
 from dbsp.project import DBSPProject
 from dbsp.error import TimeoutException
+from dbsp.connector import DBSPConnector
+
 
 class DBSPPipelineConfig:
     """Pipeline configuration specified by the user when creating
     a new pipeline instance."""
 
-    def __init__(self, project: DBSPProject, workers: int):
+    def __init__(self, project: DBSPProject, workers: int, name: str = '<anon>', description: str = ''):
         self.project = project
         self.api_client = self.project.api_client
         self.pipeline_config = PipelineConfig(
-            workers = workers,
-            inputs = PipelineConfigInputs(),
-            outputs = PipelineConfigOutputs()
+            workers=workers,
+            inputs=PipelineConfigInputs(),
+            outputs=PipelineConfigOutputs()
         )
         self.config_id = None
         self.config_version = None
-        # self.workers = workers
-        # print("config: " + str(self.pipeline_config))
+        self.attached_connectors = []
+        self.pipeline_id = None
+        self.name = name
+        self.description = description
 
-    def add_input(self, name: str, input_endpoint_config: InputEndpointConfig):
+    def add_input(self, stream: str, input_endpoint_config: DBSPConnector):
         """Add an input endpoint to the pipeline configuration.
 
         Args:
-            name (str): Endpoint name (must be unique across input endpoints).
+            stream (str): Input name (what table to connect to).
             input_endpoint_config (InputEndpointConfig): Endpoint configuration.
         """
+        input_endpoint_config.save()
 
-        # print("yaml:\n" + str(yaml.dump(input_endpoint_config.to_dict())))
-        self.pipeline_config.inputs[name] = input_endpoint_config
+        self.attached_connectors.append(AttachedConnector(
+            connector_id=input_endpoint_config.connector_id,
+            direction=Direction.INPUT,
+            uuid=uuid.uuid4().hex,
+            config=stream,
+        ))
 
-    def add_file_input(self, stream: str, filepath: str, format_: FormatConfig, **file_config_options):
+    def add_kafka_input(self, name: str, stream: str, config: KafkaInputConfig, format: FormatConfig):
         """Add an input endpoint that reads data from a file to the pipeline configuration.
 
         Args:
-            stream (str): Input stream name to connect the endpoint to.
+            name (str): Name of the input endpoint.
```

Comment:
I no longer know what the terminology means, but I guess this is connector name, not endpoint name?